[filename-case] incorrect filename reporting message

<!-- ✨ Thanks for reporting a bug! ➡️ Please don't ignore this template -->

<!-- 1️⃣ Explain here what's wrong -->

<!-- 2️⃣ Specify which rule is buggy here and in the title -->

<!-- 3️⃣ Add some examples where the issue appears -->

```js
// 1_.js
```

```sh
eslint 1_.js
```

```log
/Users/JounQin/Workspaces/GitHub/eslint-mdx/1_.js
  1:1  error  Filename is not in kebab case or pascal case. Rename it to `1.js` or `1.js`  unicorn/filename-case

✖ 1 problem (1 error, 0 warnings)
```

--- a/rules/filename-case.js
+++ b/rules/filename-case.js
@@ -91,7 +91,7 @@ function fixFilename(words, caseFunctions, {leading, extension}) {
 		samples: combinations
 	} = cartesianProductSamples(replacements);
 
-	return combinations.map(parts => `${leading}${parts.join('')}${extension}`);
+	return [...new Set(combinations.map(parts => `${leading}${parts.join('')}${extension}`))];
 }
 
 const leadingUnderscoresRegex = /^(?<leading>_+)(?<tailing>.*)$/;

--- a/test/filename-case.mjs
+++ b/test/filename-case.mjs
@@ -528,6 +528,16 @@ test({
 					ignore: [/^foo/, /^bar/]
 				}
 			]
+		),
+		// #1136
+		testManyCases(
+			'src/foo/1_.js',
+			{
+				camelCase: true,
+				pascalCase: true,
+				kebabCase: true
+			},
+			'Filename is not in camel case, pascal case, or kebab case. Rename it to `1.js`.'
 		)
 	]
 });

